mne/inverse_sparse/mxne_optim.py

```diff
@@ -930,23 +930,18 @@ def _tf_mixed_norm_solver_bcd_(M, G, Z, active_set, candidates, alpha_space,
                         active_set_j[:] = True
                         R -= np.dot(G_j, phiT(Z[jj]))
 
-        if log_objective:
-            if (ii + 1) % 10 == 0:
-                Zd = np.vstack([Z[pos] for pos in range(n_positions)
-                               if np.any(Z[pos])])
-                gap, p_obj, d_obj, _ = dgap_l21l1(
-                    M, Gd, Zd, active_set, alpha_space, alpha_time, phi, phiT,
-                    shape, n_orient, d_obj)
-                converged = (gap < tol)
-                E.append(p_obj)
-                logger.info("\n    Iteration %d :: n_active %d" % (
-                            ii + 1, np.sum(active_set) / n_orient))
-                logger.info("    dgap %.2e :: p_obj %f :: d_obj %f" % (
-                            gap, p_obj, d_obj))
-        else:
-            if (ii + 1) % 10 == 0:
-                logger.info("\n    Iteration %d :: n_active %d" % (
-                            ii + 1, np.sum(active_set) / n_orient))
+        if (ii + 1) % log_objective == 0:
```

Comment:
what if log_objective is np.inf here?

Reply:
We never enter and do not compute the dgap, similar as when log_objective=False in master
The difference is that with this PR we will not print the iteration/AS size either so user gets less feedback

Reply:
Do we need a test that log_objective > 0?

Reply:
I don't think so, if ` (ii + 1) % log_objective == 0` then ` (ii + 1) % (-log_objective)` is also 0 so it wouldn't cause trouble.